python/pyspark/ml/feature.py

```diff
@@ -790,18 +790,22 @@ class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol):
     [(0, 0.0), (1, 2.0), (2, 1.0), (3, 0.0), (4, 0.0), (5, 1.0)]
     """
 
+    handleInvalid = Param(Params._dummy(), "handleInvalid", "A handler in case of invalid column.")
+
     @keyword_only
-    def __init__(self, inputCol=None, outputCol=None):
+    def __init__(self, handleInvalid="error", inputCol=None, outputCol=None):
```

Comment:
I think we can make ```handleInvalid``` shared param just like what we do in Scala API. I have submit #8102 for this issue, please feel free to comment.